src/transaction_builder.js

```diff
@@ -234,9 +470,52 @@ export class TransactionBuilder {
       this.timebounds.maxTime.toString()
     );
 
-    attrs.cond = xdr.Preconditions.precondTime(
-      new xdr.TimeBounds(this.timebounds)
-    );
+    const timeBounds = new xdr.TimeBounds(this.timebounds);
+
+    if (this.hasV2Preconditions()) {
+      let ledgerBounds = null;
+      if (this.ledgerbounds !== null) {
+        ledgerBounds = new xdr.LedgerBounds({
+          minLedger: UnsignedHyper.fromString(
+            this.ledgerbounds.minLedger.toString()
+          ),
+          maxLedger: UnsignedHyper.fromString(
+            this.ledgerbounds.maxLedger.toString()
+          )
+        });
+      }
+
+      let minSeqNum =
+        this.minAccountSequence !== null ? this.minAccountSequence : '0';
+      minSeqNum = new xdr.SequenceNumber(UnsignedHyper.fromString(minSeqNum));
+
+      const minSeqAge = UnsignedHyper.fromString(
+        this.minAccountSequenceAge !== null
+          ? this.minAccountSequenceAge.toString()
+          : '0'
+      );
+
+      const minSeqLedgerGap = this.minAccountSequenceLedgerGap || 0;
+
+      // TODO: Parse these somehow? or make them a richer type?
+      // Re: Yep, we might need a StrKey -> xdr.SignerKey abstraction :(
```

Comment:
Well.. `StrKey` wouldn't handle signed payloads, right? Maybe instead of a `string[]` input this needs to be a `Signer[]` input?

Reply:
StrKey handles signed payloads after #511. 

In the Go SDK, we add extra methods on top of `xdr.SignerKey` to make the strkey <--> xdr.SignerKey abstraction possible (see [xdr/signer_key.go](https://github.com/stellar/go/blob/master/xdr/signer_key.go)). We can't do that in JS, really, but we can build a similar translation layer.

To keep this PR simple, though, we should just make this a `[]xdr.SignerKey` over a `[]string`. That way, we can work on the abstraction separately and squeeze it in before the final release.